public final class org.apache.http.util.ExceptionUtils extends java.lang.Object
{
    private static final java.lang.reflect.Method INIT_CAUSE_METHOD;

    static void <clinit>()
    {
        java.lang.reflect.Method $r0;

        $r0 = staticinvoke <org.apache.http.util.ExceptionUtils: java.lang.reflect.Method getInitCauseMethod()>();

        <org.apache.http.util.ExceptionUtils: java.lang.reflect.Method INIT_CAUSE_METHOD> = $r0;

        return;
    }

    private void <init>()
    {
        org.apache.http.util.ExceptionUtils r0;

        r0 := @this: org.apache.http.util.ExceptionUtils;

        specialinvoke r0.<java.lang.Object: void <init>()>();

        return;
    }

    private static java.lang.reflect.Method getInitCauseMethod()
    {
        java.lang.Class[] $r0;
        java.lang.Class $r1;
        java.lang.Throwable $r2;
        java.lang.reflect.Method r3;

     label1:
        $r1 = class "Ljava/lang/Throwable;";

        $r0 = newarray (java.lang.Class)[1];

        $r0[0] = class "Ljava/lang/Throwable;";

        r3 = virtualinvoke $r1.<java.lang.Class: java.lang.reflect.Method getMethod(java.lang.String,java.lang.Class[])>("initCause", $r0);

     label2:
        return r3;

     label3:
        $r2 := @caughtexception;

        return null;

        catch java.lang.NoSuchMethodException from label1 to label2 with label3;
    }

    public static void initCause(java.lang.Throwable, java.lang.Throwable)
    {
        java.lang.Throwable r0, r5;
        java.lang.reflect.Method r1;
        java.lang.Object[] $r2;
        java.lang.Exception $r4;

        r5 := @parameter0: java.lang.Throwable;

        r0 := @parameter1: java.lang.Throwable;

        r1 = <org.apache.http.util.ExceptionUtils: java.lang.reflect.Method INIT_CAUSE_METHOD>;

        if r1 == null goto label4;

     label1:
        $r2 = newarray (java.lang.Object)[1];

        $r2[0] = r0;

        virtualinvoke r1.<java.lang.reflect.Method: java.lang.Object invoke(java.lang.Object,java.lang.Object[])>(r5, $r2);

     label2:
        goto label4;

     label3:
        $r4 := @caughtexception;

     label4:
        return;

        catch java.lang.Exception from label1 to label2 with label3;
    }
}
